libp2p/protocol_muxer/multiselect_communicator.py

```diff
@@ -11,9 +14,26 @@ def __init__(self, read_writer: ReadWriteCloser) -> None:
         self.read_writer = read_writer
 
     async def write(self, msg_str: str) -> None:
+        """
+        :raise MultiselectCommunicatorError: raised when failed to write to underlying reader
+        """
         msg_bytes = encode_delim(msg_str.encode())
-        await self.read_writer.write(msg_bytes)
+        try:
+            await self.read_writer.write(msg_bytes)
+        except IOException:
+            raise MultiselectCommunicatorError(
+                "fail to write to multiselect communicator"
+            )
 
     async def read(self) -> str:
-        data = await read_delim(self.read_writer)
+        """
+        :raise MultiselectCommunicatorError: raised when failed to read from underlying reader
+        """
+        try:
+            data = await read_delim(self.read_writer)
+        # `IOException` includes `IncompleteReadError` and `StreamError`
+        except (ParseError, IOException, ValueError):
```

Comment:
Note: `ValueError` seems more appropriate to be replaced by other errors like `IOException`.

Reply:
Ah it seems `ValueError` is removed from `read_delim`?